extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTaskClient.java

```diff
@@ -143,147 +111,147 @@ public void resume(TaskLocation location, String id)
-
+      final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "offsets/current", null, retry);
       return jsonMapper.readValue(response.getContent(), new TypeReference<Map<Integer, Long>>() {});
     }
-    catch (Exception e) {
+    catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public Map<Integer, Long> getEndOffsets(TaskLocation location, String id)
+  public Map<Integer, Long> getEndOffsets(String id)
   {
     try {
-      final URI serviceUri = new URI(
-          "http",
-          null,
-          location.getHost(),
-          location.getPort(),
-          String.format("%s/%s/%s", BASE_PATH, id, "offsets/end"),
-          null,
-          null
-      );
-      final StatusResponseHolder response = submitRequest(new Request(HttpMethod.GET, serviceUri.toURL()), true);
-
+      final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "offsets/end", null, true);
       return jsonMapper.readValue(response.getContent(), new TypeReference<Map<Integer, Long>>() {});
     }
-    catch (Exception e) {
+    catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public void setEndOffsets(TaskLocation location, String id, Map<Integer, Long> endOffsets)
+  public void setEndOffsets(String id, Map<Integer, Long> endOffsets)
   {
-    setEndOffsets(location, id, endOffsets, false);
+    setEndOffsets(id, endOffsets, false);
   }
 
-  public void setEndOffsets(TaskLocation location, String id, Map<Integer, Long> endOffsets, boolean resume)
+  public void setEndOffsets(String id, Map<Integer, Long> endOffsets, boolean resume)
   {
     try {
-      final URI serviceUri = new URI(
-          "http",
-          null,
-          location.getHost(),
-          location.getPort(),
-          String.format("%s/%s/%s", BASE_PATH, id, "offsets/end"),
-          resume ? "resume=true" : null,
-          null
-      );
       submitRequest(
-          new Request(HttpMethod.POST, serviceUri.toURL()).setContent(
-              MediaType.APPLICATION_JSON,
-              jsonMapper.writeValueAsBytes(endOffsets)
-          ), true
+          id,
+          HttpMethod.POST,
+          "offsets/end",
+          resume ? "resume=true" : null,
+          jsonMapper.writeValueAsBytes(endOffsets),
+          true
       );
     }
-    catch (Exception e) {
+    catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  private StatusResponseHolder submitRequest(Request request, boolean retry)
+  @VisibleForTesting
+  RetryPolicyFactory createRetryPolicyFactory() {
+    return new RetryPolicyFactory(new RetryPolicyConfig()
+                                      .setMinWait(Period.seconds(2))
+                                      .setMaxWait(Period.seconds(8))
+                                      .setMaxRetryCount(5)
+    );
+  }
+
+  @VisibleForTesting
+  void checkConnection(String host, int port) throws IOException
+  {
+    new Socket(host, port).close();
```

Comment:
This is intended to throw if it fails so it can be handled by the catch block in submitRequest() with the retry logic in it. There's a comment at the point this method is called which describes why this code is here - it's basically a fail fast to stop us from trying to talk to a non-responsive task through Netty which generates a lot of scary log messages when the connection fails which happen quite regularly. Why do connections fail? When a new task is created, it is reported as 'running' before the Jetty server is fully initialized so we try to start talking to it and get no response back. This is handled by the retry logic, but I didn't want a bunch of scary looking exceptions in the logs when nothing is out of the ordinary. submitRequest() will log some debug level logs when this call throws an exception.